app/validators/checksum_validator.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+class ChecksumValidator < ActiveModel::Validator
+  include Rails.application.routes.url_helpers
+  def validate(record)
+    @record = record
+    @uploaded_file = @record.file
+    @duvs_obj = DuplicateUploadVerificationService.new(@uploaded_file)
+    if !@duvs_obj.duplicates.empty?
+      record.errors.add(:checksum, in_solr_error_message)
+    elsif Sufia::UploadedFile.begun_ingestion.map(&:checksum).include?(@record.checksum)
+      record.errors.add(:checksum, begun_ingestion_error_message)
+    elsif Sufia::UploadedFile.where(batch_upload_id: @record.batch_upload_id).map(&:checksum).include?(@record.checksum)
```

Comment:
We can probably split these 3 validations up into separate methods.

Reply:
@RudyOnRails yes. It's hard to see the logic elsifs because the chain methods are very long. What would it look like if you could rewrite it as a case statement? Could you create an internal class the encapsulated all the where logic and simply reported the type of error?